modules/user_accounts/php/edit_user.class.inc

```diff
@@ -454,12 +454,27 @@ class Edit_User extends \NDB_Form
             $user    = \User::factory($this->identifier);
             $success = $user->update($set);
         }
-
         // Now set the password. Note that this field is named incorrectly
         // and represents a plaintext password, not a hash.
-        if (isset($values['Password_hash'])) {
-            $user->updatePassword($values['Password_hash']);
+        if (isset($values['Password_hash'], $values['Password_expiry'])) {
```

Comment:
The third param for updatePassword must be a `?\DateTime` so `$input_password_expiry` needs to be a DateTime object. However, doing `new DateTime(null);` gives today's date instead of throwing any kind of error.